Automattic/jetpack
# Update all occurrences of MJ utils/formatNumber() function to use default options.

In PR https://github.com/Automattic/jetpack/pull/38953, we updated the My Jetpack `formatNumber()` [utility function](https://github.com/Automattic/jetpack/blob/8a135aff685d3fc79f990d4982de5507ed8b8a99/projects/packages/my-jetpack/_inc/utils/format-number.ts) to pass some custom properties to the 2nd "options" parameter of the function by default. The custom default options are:
```
{
	maximumFractionDigits: 1,
	notation: 'compact',
}
```
These make the `formatNumber()` function display compact numbers (with 1 decimal) by default. For example: 23,682 displays as `23.7K`, or 1,212,000 as `1.2M`. This was our main intention/use case for the function.

Therefore, this is a follow-up PR: that updates all the other places that we call the My Jetpack formatNumber() function, removing the option parameters we were passing in because they are no longer needed (because as stated above, we now pass them in by default in the function definition).






Fixes #

## Proposed changes:

* Update occurrences of formatNumber() in the My Jetpack Backup product card.
* Update occurrences of formatNumber in the My Jetpack VideoPress product card.
* Update occurrences of FormatNumber in the My Jetpack Stats section.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* It's probably enough just to review the code and verify that I have removed the 2nd argument (`shortenedNumberConfig`) on each instance where we are calling the My Jetpack `numberFormat` utility function.
    - You can view the [my-jetpack/_inc/utils/format-number.ts](https://github.com/Automattic/jetpack/blob/8a135aff685d3fc79f990d4982de5507ed8b8a99/projects/packages/my-jetpack/_inc/utils/format-number.ts) function definition and see that the "shortenedNumberConfig" options are now passed into the function by default.

If you want to do a more thorough manual test, follow the following steps below:
- Checkout this branch on your local Jetpack dev site, or a JN site using Beta plugin with sandbox access enabled.
- Connect your site and make your Jetpack site is using your sandbox (`JETPACK_SANDBOX__DOMAIN`).
- In your sandbox, edit file: `public.api/rest/wpcom-json-endpoints/class.wpcom-json-api-stats-highlights-v1-1-endpoint.php`
    - Disable the cache by commenting out line 22 (`// return $cached_response;`)
    - Line 34, set views to 8100. (`'views' => 8100,`)
- Open the "Stats detail page" by going to My Jetpack --> Stats card --> View detailed stats, or directly to: `/wp-admin/admin.php?page=stats`
- Now go back to My Jetpack page.
- View the Stats Highlights, the "Views" card should be showing `8.1k` (shortened number format), but the difference should Not have the shortened format (because the value is less than 9999).  See screenshot:
![Screen Shot 2024-08-27 at 16 52 40](https://github.com/user-attachments/assets/98070475-d3e8-4541-a2c0-6e6f08ff7892)
- Again in your sandbox, edit file: `public.api/rest/wpcom-json-endpoints/class.wpcom-json-api-stats-highlights-v1-1-endpoint.php`
    - Line 34, set views to 10100. (`'views' => 10100,`)
- Open the Stats detail page by going to My Jetpack --> Stats card --> View detailed stats, or directly to `/wp-admin/admin.php?page=stats`
- Now go back to My Jetpack page.
- View the Stats Highlights, the "Views" card should be showing `10.1k` (shortened number format), and also the difference should have the shortened format also (because the value is greater than 9999).  See screenshot:
![Screen Shot on 2024-08-27 at 16-45-51](https://github.com/user-attachments/assets/c00bcef9-0885-4b52-aba4-e9c183f8cab6)
